src/Compilers/CSharp/Test/Semantic/Semantics/UsingDeclarationTests.cs

```diff
@@ -0,0 +1,396 @@
+                // (10,9): error CS8641: A goto target within the same block can not cross a using declaration.
+                //         goto label1;
+                Diagnostic(ErrorCode.ERR_GoToBackwardJumpOverUsingVar, "goto label1;").WithLocation(10, 9)
+                );
+        }
+
+        [Fact]
+        public void AllowGoToAroundUsingDeclarations()
+        {
+            var source = @"
+using System;
+class C
+{
+    static void Main()
+    {
+        goto label1;
+        label1:
+        using var x = (IDisposable)null;
+        goto label2;
+        label2:
+        return;
+    }
+}
+";
+            CreateCompilation(source).VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void AllowGoToAcrossUsingDeclarationsBackwardsWhenFromADifferentBlock()
+        {
+            var source = @"
+using System;
+class C
+{
+    static void Main()
+    {
+        label1:
+        using var x = (IDisposable)null;
+        {
+            if (true)
+            {
+                goto label1;
+            }
+        }
+    }
+}
+";
+            CreateCompilation(source).VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void UsingVariableCanBeInitializedWithExistingDisposable()
+        {
+            var source = @"
+using System;
+class C2
+{
+    public void Dispose()
+    {
+        Console.Write(""Disposed; ""); 
+    }
+}
+class C
+{
+    static void Main()
+    {
+        var x = new C2();
+        using var x2 = x;
+        using var x3 = x2;
+        x = null;
+    }
+}
+";
+            var compilation = CreateCompilation(source, options: TestOptions.DebugExe).VerifyDiagnostics();
+            CompileAndVerify(compilation, expectedOutput: "Disposed; Disposed; ");
+        }
+
+        [Fact]
+        public void UsingVariableCanBeInitializedWithExistingDisposableInASingleStatement()
+        {
+            var source = @"
+using System;
+class C2
+{
+    public void Dispose()
+    {
+        Console.Write(""Disposed; ""); 
+    }
+}
+class C
+{
+    static void Main()
+    {
+        using C2 x = new C2(), x2 = x;
+    }
+}
+";
+            var compilation = CreateCompilation(source, options: TestOptions.DebugExe).VerifyDiagnostics();
+            CompileAndVerify(compilation, expectedOutput: "Disposed; Disposed; ");
+        }
+
+        [Fact]
+        public void UsingVariableCannotBeReAssigned()
+        {
+            var source = @"
+using System;
+class C
+{
+    static void Main()
+    {
+        using var x = (IDisposable)null;
+        x = null;
+    }
+}
+";
+            CreateCompilation(source).VerifyDiagnostics(
+                // (8,9): error CS1656: Cannot assign to 'x' because it is a 'using variable'
+                //         x = null;
+                Diagnostic(ErrorCode.ERR_AssgReadonlyLocalCause, "x").WithArguments("x", "using variable").WithLocation(8, 9)
+                );
+        }
+
+        [Fact]
+        public void VariableCannotBeReAssignedAsUsingVariable()
```

Comment:
>VariableCannotBeReAssignedAsUsingVariable [](start = 20, length = 41)

Is the test name correct? And perhaps remove the test unless it is testing a distinct scenario.